modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,130 @@
+		return
+	if (mutant_variants & STYLE_TAUR)
+		return
+	if (has_taur_sprite[icon_state])
+		if (has_taur_snake_sprite[icon_state] && (!(mutant_variants & STYLE_TAUR_SNAKE)))
+			mutant_variants |= STYLE_TAUR_SNAKE
+		if (has_taur_horse_sprite[icon_state] && (!(mutant_variants & STYLE_TAUR_HOOF)))
+			mutant_variants |= STYLE_TAUR_HOOF
+		if (has_taur_paw_sprite[icon_state] && (!(mutant_variants & STYLE_TAUR_PAW)))
+			mutant_variants |= STYLE_TAUR_PAW
+		return // If we already know this icon state has a taur sprite, skip the for loop and take from the cache
+	else if (has_taur_sprite[icon_state] == HAS_NO_TAUR_SPRITE)
+		return
+	// Code only goes here if has_taur_sprite[] == nothing, AKA if init has never been ran
+	var/taur_sprite = FALSE
+	if (icon_state in GLOB.naga_taur_uniform_sprites)
+		has_taur_sprite[icon_state] = HAS_TAUR_SPRITE
+		has_taur_snake_sprite[icon_state] = HAS_TAUR_SPRITE
+		taur_sprite = TRUE
+		if (!(mutant_variants & STYLE_TAUR_SNAKE))
+			mutant_variants |= STYLE_TAUR_SNAKE
+	if (icon_state in GLOB.horse_taur_uniform_sprites)
+		has_taur_sprite[icon_state] = HAS_TAUR_SPRITE //This block of code is checking the 3 DMI files for the icon state and setting flags/vars dynamically based on that
+		has_taur_horse_sprite[icon_state] = HAS_TAUR_SPRITE // We want to avoid using this because for loops on init are costly-hence, the lists we use
+		taur_sprite = TRUE
+		if (!(mutant_variants & STYLE_TAUR_HOOF))
+			mutant_variants |= STYLE_TAUR_HOOF
+	if (icon_state in GLOB.pawed_taur_uniform_sprites)
+		has_taur_sprite[icon_state] = HAS_TAUR_SPRITE
+		has_taur_paw_sprite[icon_state] = HAS_TAUR_SPRITE
+		taur_sprite = TRUE
+		if (!(mutant_variants & STYLE_TAUR_PAW))
+			mutant_variants |= STYLE_TAUR_PAW
+	if (!taur_sprite) // If none of the 3 above if statements are true, it has no alt sprite
+		has_taur_sprite[icon_state] = HAS_NO_TAUR_SPRITE
+
+
+/**
+ * Called in handle_taur_sprites().
+ *
+ * Suit variation of handle_taur_sprites().
+ *
+ * Quits instantly if mutant_variants has all taur styles (STYLE_TAUR), then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite_suit is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing if the
+ * clothing doesnt already have it.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite_suit for that icon state will be set to false.
+ *
+ * DO NOT RELY ON THIS PROC TO DO YOUR WORK FOR YOU. If you are adding a new item, or a sprite, you SHOULD be manually setting your mutant_variants flags, otherwise the for
+ * loop may be ran and cause slight performance loss.
+ */
+/obj/item/clothing/proc/handle_taur_sprites_for_suits()
```

Comment:
I'm really curious to see how much of an effect they have on performance, I feel like it won't be great.

Reply:
Great as in, it won't be too taxing or it WILL be taxing?